mindx-login/src/components/UserProvider.jsx

```diff
@@ -0,0 +1,35 @@
+import { createContext, useContext, useState, useEffect } from "react";
+
+const UserContext = createContext(null);
+const SetUserContext = createContext(null);
+
+export const useUser = () => {
+  return useContext(UserContext);
+};
+
+export const useSetUser = () => {
+  return useContext(SetUserContext);
+};
+
+const UserProvider = ({ children }) => {
+  const [user, setUser] = useState(
+    JSON.parse(localStorage.getItem("mindx-user")) || {
+      name: "",
+      auth: false,
+    }
+  );
+
+  useEffect(() => {
+    localStorage.setItem("mindx-user", JSON.stringify(user));
+  }, [user])
+
+  return (
+    <UserContext.Provider value={user}>
+      <SetUserContext.Provider value={setUser}>
+        {children}
+      </SetUserContext.Provider>
+    </UserContext.Provider>
+  );
```

Comment:
Em dùng 1 Context và 1 Provider thôi, khong cần phải làm 1 Provider như vậy, mình có thể truyền 1 object nhiều property vào provider value  
```suggestion
    <UserContext.Provider value={{user, setUser}}>
        {children}
    </UserContext.Provider>
  );
```